### DISCUSSION TITLE
Query on dependency versioning

### DISCUSSION THREAD

**Discussion by DavidGerald-Milligan** [junit-team/junit-framework#4214] [Q&A]
When I add maven dependencies on _org.junit.jupiter:junit-jupiter-api:jar:5.11.3_  and _org.junit.jupiter:junit-jupiter-engine:jar:5:11.3_ and invoke assertThrows I get _java.lang.NoClassDefFoundError: org/junit/platform/commons/util/UnrecoverableExceptions_

UnrecoverableExceptions is present in _org.junit.platform:junit-platform-commons:5.11.3_ and [mvnrepository.com](https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-api/5.11.3) lists this version as a dependency.

Yet when I query the dependency tree I get

[INFO] \- org.junit.jupiter:junit-jupiter-api:jar:5.11.3:test
[INFO]    +- org.opentest4j:opentest4j:jar:1.3.0:test
[INFO]    +- org.junit.platform:junit-platform-commons:jar:1.6.2:test
[INFO]    \- org.apiguardian:apiguardian-api:jar:1.1.2:test

_org.junit.platform:junit-platform-commons:jar:1.6.2_ does not include UnrecoverableExceptions 

Updating my dependencies to _org.junit.jupiter:junit-jupiter:jar:5.11.3_ and _org.junit.jupiter:junit-jupiter-api:jar:5.11.3_ resolves the issue but I am trying to understand why we have the difference in the tree and the documented dependencies.

**Comment by sormuras** [comment]
> Unless you’re using Spring Boot which defines its own way of managing dependencies, it is recommended to use the JUnit Platform BOM to align the versions of all JUnit 5 artifacts.

https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven-bom

> [...] why we have the difference in the tree and the documented dependencies.

Perhaps some other dependency already refers to platform-commons and "locks the version to be used to 1.6.2"?

**Reply by mpkorstanje** [reply]
@DavidGerald-Milligan you can use `mvn dependency:tree` with `-Dverbose` to get a bit more info about the way maven up/downgrades dependencies.
